Libraries/Components/View/ReactNativeStyleAttributes.windows.js

```diff
@@ -0,0 +1,48 @@
+/**
```

Comment:
>/ [](start = 0, length = 1)

UWP has some named acrylic options (you can see some at [https://docs.microsoft.com/en-us/windows/uwp/style/acrylic](https://docs.microsoft.com/en-us/windows/uwp/style/acrylic)). Can you give an example of how to represent SystemControlAcrylicWindowMediumHighBrush using this API you've exposed?

Reply:
I'm not sure if it's a good idea to expose XAML resources thru RN interface, but if you think otherwise I can add another property for that.

Reply:
As discussed offline, what I meant was that MSDN describes how to use acrylic in terms of recommended brushes like `SystemControlAcrylicWindowMediumHighBrush`. So when React Native users come across this documentation and decide they want to use a recommended acrylic config from MSDN, how do they apply that knowledge with React Native?

Maybe they go look up the parameters of the brush and use the parameters with the `acrylic*` props you've added.

Or maybe React Native should come with some acrylic presets with parameters hardcoded in JavaScript.

Or maybe React Native should allow users to reference XAML brushes.

One thing to consider is if there's a benefit to using XAML brushes versus hardcoding the acrylic parameters in JavaScript such that they match the brushes.

Reply:
This problem is not specific to acrylic brushes as there are other theme-specific brushes in XAML. My impression is that we need a way for JS to list the available theme-brushes and assign one of them to a react element (we'll need a new style for that, maybe `background`):

```js
RN.getThemeBrushes().then(brushes => {
  if (brushes.SystemControlAcrylicWindowMediumHighBrush) {
    ...;
  } else {
    ...;
  }
});
```

This interface looks clumsy, though, because if JS wants to use a specific brush, e.g. that SystemControlAcrylicWindowMediumHighBrush, and the brush doesn't exist because the app is running on RS2, JS needs to come up with a set of fallback styles. Perhaps this could be achieved by specifying both `background` and the usual set of `backgroundColor` etc. styles, so if the brush doesn't exist, RNW would use the fallback styles:

```js
RX.Styles.createViewStyle({
        backgroundUWP: 'SystemControlAcrylicWindowMediumHighBrush',
        backgroundColor: '#ffe', // fallback style
        flexDirection: 'row',
        paddingVertical: 8,
        justifyContent: 'space-between',
        overflow: 'visible',
});
```

We need @rozele here.

Reply:
Instead of `RN.getThemeBrushes()`, could we export the brushes synchronously during initialization using the `Constants` method?

How do native XAML apps handle the case of attempting to use a brush on RS2 that doesn't exist until RS3?

Implementation-wise, would `brushes.SystemControlAcrylicWindowMediumHighBrush` literally reference the XAML brush on the native side or would `brushes.SystemControlAcrylicWindowMediumHighBrush` just contain the appropriate config (e.g. opacity, color) to mimic the XAML brush?

Reply:
I think those XAML apps have to use some conditional markup or some C# code for that. There is a [discussion](https://github.com/Windows-XAML/Template10/issues/1521) related to this which says the following about XAML: _"This only has to do with our controls, not the framework. Still. I hate that XAML is not backwards compatible. The good news is that XAML in RS3 is incompatible with RS2 and RS1, but it introduces a mechanism to keep it from breaking in the future. That is to say, it is becoming backwards compatible."_

Reply:
Can you find out the recommended pattern for native XAML apps? Maybe check with George or somebody working on Fluent.